src/App.jsx

```diff
@@ -14,45 +16,142 @@ export const goodsFromServer = [
+const REVERSE = 'Reverse';
+
+function getPreparedGoods(goods, sortFiled, reverse) {
+  const preparedGoods = [...goods];
+
+  if (sortFiled) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortFiled) {
+        case ALPHABET:
+          return good1.localeCompare(good2);
+
+        case LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse === REVERSE) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  // const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortFiled, setSortFiled] = useState('');
+  const [reverse, setReverse] = useState('');
+
+  let visibleGoods = getPreparedGoods(goodsFromServer, sortFiled, reverse);
+
+  // function sortAlphabetically() {
+  //   if (reverse === REVERSE) {
+  //     setVisibleGoods([...visibleGoods].sort().reverse());
+  //   } else {
+  //     setVisibleGoods([...visibleGoods].sort());
+  //   }
+
+  //   setSortFiled(ALPHABET);
+  // }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  // function sortByLength() {
+  //   if (reverse === REVERSE) {
+  //     setVisibleGoods([...visibleGoods].sort(
+  //       (good1, good2) => good2.length - good1.length,
+  //     ));
+  //   } else {
+  //     setVisibleGoods([...visibleGoods].sort(
+  //       (good1, good2) => good1.length - good2.length,
+  //     ));
+  //   }
+
+  //   setSortFiled(LENGTH);
+  // }
+
+  function reverseSort() {
+    [...visibleGoods].reverse();
+    if (reverse === '') {
+      setReverse(REVERSE);
+    } else {
+      setReverse('');
+    }
+  }
+
+  function reset() {
+    visibleGoods = goodsFromServer;
+    setSortFiled('');
+    setReverse('');
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortFiled(ALPHABET)}
+          type="button"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortFiled !== ALPHABET },
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortFiled(LENGTH)}
+          type="button"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortFiled !== LENGTH },
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={reverseSort}
+          type="button"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': reverse !== REVERSE },
+          )}
+        >
+          Reverse
+        </button>
+
+        {JSON.stringify(goodsFromServer) !== JSON.stringify(visibleGoods)
```

Comment:
just check either we have sort field and reversed value is true to display this button